[CI] ProfileCancellationIntegTests testSuggestProfilesCancellation failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/k3titqku4p2be/tests/:x-pack:plugin:security:internalClusterTest/org.elasticsearch.xpack.security.profile.ProfileCancellationIntegTests/testSuggestProfilesCancellation

**Reproduction line:**
`./gradlew ':x-pack:plugin:security:internalClusterTest' --tests "org.elasticsearch.xpack.security.profile.ProfileCancellationIntegTests.testSuggestProfilesCancellation" -Dtests.seed=74F6D1B80FACB410 -Dtests.locale=no-NO -Dtests.timezone=BET -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.security.profile.ProfileCancellationIntegTests&tests.test=testSuggestProfilesCancellation

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[107]>

  at __randomizedtesting.SeedInfo.seed([74F6D1B80FACB410:B026926B1FBFB102]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.xpack.security.profile.ProfileCancellationIntegTests.lambda$testSuggestProfilesCancellation$3(ProfileCancellationIntegTests.java:119)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1096)
  at org.elasticsearch.xpack.security.profile.ProfileCancellationIntegTests.testSuggestProfilesCancellation(ProfileCancellationIntegTests.java:109)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileCancellationIntegTests.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileCancellationIntegTests.java
@@ -22,6 +22,7 @@ import org.elasticsearch.plugins.ActionPlugin;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.plugins.PluginsService;
 import org.elasticsearch.search.internal.ReaderContext;
+import org.elasticsearch.search.internal.SearchContext;
 import org.elasticsearch.tasks.CancellableTask;
 import org.elasticsearch.tasks.Task;
 import org.elasticsearch.transport.TransportService;
@@ -101,7 +102,11 @@ public class ProfileCancellationIntegTests extends AbstractProfileIntegTestCase 
                 taskActions,
                 hasItems(equalTo(SuggestProfilesAction.NAME), equalTo(SearchAction.NAME), startsWith(SearchAction.NAME))
             );
-            tasks.forEach(t -> taskIds.add(t.getId()));
+            assertThat(isShardSearchBlocked(), is(true));
+            tasks.forEach(t -> {
+                logger.info("task " + t.getId() + "/" + t.getAction() + " registered");
+                taskIds.add(t.getId());
+            });
         }, 20, TimeUnit.SECONDS);
 
         // Cancel the suggest request and all tasks should be cancelled
@@ -166,11 +171,21 @@ public class ProfileCancellationIntegTests extends AbstractProfileIntegTestCase 
         }
     }
 
+    private boolean isShardSearchBlocked() {
+        for (PluginsService pluginsService : internalCluster().getInstances(PluginsService.class)) {
+            if (pluginsService.filterPlugins(SearchBlockPlugin.class).stream().anyMatch(SearchBlockPlugin::isShardSearchBlocked)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     public static class SearchBlockPlugin extends Plugin implements ActionPlugin {
         protected static final Logger logger = LogManager.getLogger(SearchBlockPlugin.class);
 
         private final String nodeId;
         private final AtomicBoolean shouldBlockOnSearch = new AtomicBoolean(false);
+        private final AtomicBoolean shardSearchBlocked = new AtomicBoolean(false);
 
         public SearchBlockPlugin(Settings settings, Path configPath) throws Exception {
             nodeId = settings.get("node.name");
@@ -180,12 +195,20 @@ public class ProfileCancellationIntegTests extends AbstractProfileIntegTestCase 
         public void onIndexModule(IndexModule indexModule) {
             super.onIndexModule(indexModule);
             indexModule.addSearchOperationListener(new SearchOperationListener() {
+
                 @Override
-                public void onNewReaderContext(ReaderContext readerContext) {
+                public void onPreQueryPhase(SearchContext c) {
+                    logger.info("onPreQueryPhase");
+                }
+
+                @Override
+                public void onNewReaderContext(ReaderContext c) {
                     try {
                         logger.info("blocking search on " + nodeId);
-                        assertBusy(() -> assertFalse(shouldBlockOnSearch.get()));
+                        shardSearchBlocked.set(true);
+                        assertBusy(() -> assertFalse(shouldBlockOnSearch.get()), 20, TimeUnit.SECONDS);
                         logger.info("unblocking search on " + nodeId);
+                        shardSearchBlocked.set(false);
                     } catch (Exception e) {
                         throw new RuntimeException(e);
                     }
@@ -199,6 +222,10 @@ public class ProfileCancellationIntegTests extends AbstractProfileIntegTestCase 
 
         void disableSearchBlock() {
             shouldBlockOnSearch.set(false);
+        }
+
+        boolean isShardSearchBlocked() {
+            return shardSearchBlocked.get();
         }
     }
 }

